comp/otelcol/otlp/components/statsprocessor/agent.go

```diff
@@ -0,0 +1,176 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package statsprocessor
+
+import (
+	"context"
+	"net/http"
+	"runtime"
+	"sync"
+	"time"
+
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace"
+	"github.com/DataDog/datadog-agent/pkg/trace/agent"
+	"github.com/DataDog/datadog-agent/pkg/trace/api"
+	traceconfig "github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/stats"
+	"github.com/DataDog/datadog-agent/pkg/trace/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/trace/timing"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// TraceAgent specifies a minimal trace agent instance that is able to process traces and output stats.
+type TraceAgent struct {
+	*agent.Agent
+
+	// pchan specifies the channel that will be used to output Datadog Trace Agent API Payloads
+	// resulting from ingested OpenTelemetry spans.
+	pchan chan *api.Payload
+
```

Comment:
Can we make stats processor a component ? We can we have two implementations one with `pkg/trace` and other with new interface we added for stats API ?

Reply:
That can be in a follow-up PR. Also it won't be of any use to upstream Datadog connector.